drivers/serial/uart_mcux_flexcomm.c

```diff
@@ -342,6 +379,636 @@ static int mcux_flexcomm_uart_config_get(const struct device *dev,
+	data->rx_data.next_xfer_buf = buf;
+	data->rx_data.next_xfer_len = len;
+
+	return 0;
+}
+
+/* This callback is from the TX DMA and consumed by this driver */
+static void mcux_flexcomm_uart_dma_tx_callback(const struct device *dma_device, void *cb_data,
+				uint32_t channel, int status)
+{
+	/* DMA callback data was configured during driver init as UART device ptr */
+	struct device *dev = (struct device *)cb_data;
+
+	const struct mcux_flexcomm_config *config = dev->config;
+	struct mcux_flexcomm_data *data = dev->data;
+
+	unsigned int key = irq_lock();
+
+	/* Turn off requests since we are aborting */
+	USART_EnableTxDMA(config->base, false);
+
+	/* Timeout did not happen */
+	(void)k_work_cancel_delayable(&data->tx_data.timeout_work);
+
+	irq_unlock(key);
+}
+
+/* This callback is from the RX DMA and consumed by this driver */
+static void mcux_flexcomm_uart_dma_rx_callback(const struct device *dma_device, void *cb_data,
+				uint32_t channel, int status)
+{
+	/* DMA callback data was configured during driver init as UART device ptr */
+	struct device *dev = (struct device *)cb_data;
+
+	const struct mcux_flexcomm_config *config = dev->config;
+	struct mcux_flexcomm_data *data = dev->data;
+
+	/* Cancel timeout now that the transfer is complete */
+	(void)k_work_cancel_delayable(&data->rx_data.timeout_work);
+
+	/* Update user with received RX data if needed */
+	flexcomm_uart_rx_update(dev);
+
+	/* Release current buffer */
+	struct uart_event current_buffer_release_event = {
+		.type = UART_RX_BUF_RELEASED,
+		.data.rx_buf.buf = data->rx_data.xfer_buf,
+	};
+
+	async_user_callback(dev, &current_buffer_release_event);
+
+	if (data->rx_data.next_xfer_buf) {
+		/* Replace buffer in driver data */
+		data->rx_data.xfer_buf = data->rx_data.next_xfer_buf;
+		data->rx_data.xfer_len = data->rx_data.next_xfer_len;
+		data->rx_data.next_xfer_buf = NULL;
+		data->rx_data.next_xfer_len = 0;
+
+		/* Reload DMA channel with new buffer */
+		data->rx_data.active_block.block_size = data->rx_data.xfer_len;
+		data->rx_data.active_block.dest_address = (uint32_t) data->rx_data.xfer_buf;
+		dma_reload(config->rx_dma.dev, config->rx_dma.channel,
+				data->rx_data.active_block.source_address,
+				data->rx_data.active_block.dest_address,
+				data->rx_data.active_block.block_size);
+
+		/* Request next buffer */
+		struct uart_event rx_buf_request = {
+			.type = UART_RX_BUF_REQUEST,
+		};
+
+		async_user_callback(dev, &rx_buf_request);
+
+		/* Start the new transfer */
+		dma_start(config->rx_dma.dev, config->rx_dma.channel);
+
+	} else {
+		/* If there is no next available buffer then disable DMA */
+		mcux_flexcomm_uart_rx_disable(dev);
+	}
+
+	/* Now that this transfer was finished, reset tracking variables */
+	data->rx_data.count = 0;
+	data->rx_data.offset = 0;
+}
+
+#if defined(CONFIG_SOC_SERIES_IMX_RT5XX) || defined(CONFIG_SOC_SERIES_IMX_RT6XX)
+static uint32_t fc_uart_calc_inmux_connection(uint8_t channel, DMA_Type *base)
```

Comment:
A comment documenting that this function calculates the required inputmux connection for a given channel would probably be useful for future developers

Reply:
that's the name of the function, I can un-abbreviate